src/auto-evo/mutation_strategy/UpgradeOrganelle.cs

```diff
@@ -47,7 +62,15 @@ public UpgradeOrganelle(Func<OrganelleDefinition, bool> criteria, IComponentSpec
                 {
                     // TODO: Once this is used with an upgrade that costs MP this will need to factor that in
                     organelle.Upgrades ??= new OrganelleUpgrades();
-                    organelle.Upgrades.CustomUpgradeData = upgrade;
+                    if (customUpgrade != null)
+                    {
+                        organelle.Upgrades.CustomUpgradeData = customUpgrade;
+                    }
+
+                    if (upgradeName != null)
+                    {
+                        organelle.Upgrades.UnlockedFeatures.Add(upgradeName);
```

Comment:
This should check if the unlocked features already contains the upgrade and not add it if it is there already. This is because the UnlockedFeatures is a list (and not a set that would automatically prevent this).

Reply:
Arguably it could be a set. I didn't make it initially a set for a few reasons:
- It's simpler to just have a list
- There could be save complications with a set as I'm pretty sure we don't have any sets that are deserialized yet
- Most of unlock feature lists are at most 2-3 items so always looping the entire thing is probably not any slower than a set
- There could be some performance overhead of a set, though I suppose be basically would only need to call `.Contains` most of the time

Checking all of those factors will be a bit of work which is why I haven't bothered changing it whenever I've again worked on the unlocked features variable.